magik-checks/src/main/java/nl/ramsolutions/sw/checks/moduledef/RequiredModuleAlreadyInTestsModulesCheck.java

```diff
@@ -0,0 +1,44 @@
+package nl.ramsolutions.sw.checks.moduledef;
+
+import com.sonar.sslr.api.AstNode;
+import java.util.List;
+import java.util.Set;
+import nl.ramsolutions.sw.checks.ModuleDefCheck;
+import nl.ramsolutions.sw.moduledef.api.ModuleDefinitionGrammar;
+import org.sonar.check.Rule;
+
+/** Check that no module listed in REQUIRES also appears in TESTS_MODULES. */
+@Rule(key = RequiredModuleAlreadyInTestsModulesCheck.CHECK_KEY)
+public class RequiredModuleAlreadyInTestsModulesCheck extends ModuleDefCheck {
```

Comment:
Please keep the naming consistent with other checks, to make it more apparent that it is a check for `module.def` files, as your colleague requested.